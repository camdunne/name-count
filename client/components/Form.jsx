import React from 'react';
import PropTypes from 'prop-types';

const Form = ({ handleSubmit, handleChange, value }) => (
  <center>
    <form onSubmit={handleSubmit}>
      <label htmlFor="xmlfile">
        {'Enter an xml file:  '}
        <input
          type="text"
          name="xmlfile"
          value={value}
          onChange={handleChange}
        />
      </label>
      <button id="submit" type="submit" value="Submit" />
    </form>
  </center>
);
Form.defaultProps = {
  value: '',
};
Form.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  value: PropTypes.string,
};

export default Form;
