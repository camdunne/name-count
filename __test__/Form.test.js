import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Form from '../client/components/Form';

describe('<Form />', () => {
  const defaultProps = {
    handleChange: jest.fn(() => true),
    handleSubmit: jest.fn(() => true),
    value: '',
  };
  it('renders correctly', () => {
    const tree = renderer.create(<Form {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('fires form submit', () => {
    const wrapper = mount(<Form {...defaultProps} />);
    wrapper.find('form').simulate('submit');
    expect(defaultProps.handleSubmit).toBeCalled();
  });
  it('responds to a name change', () => {
    const event = { target: { value: 'www.cool.xml' } };
    const wrapper = mount(<Form {...defaultProps} />);
    wrapper.update();
    wrapper.find('input').simulate('change', event);
    expect(defaultProps.handleChange).toBeCalled();
  });
  it('should have a value input and a submit input', () => {
    const wrapper = shallow(<Form {...defaultProps} />);
    console.log(expect(wrapper.find('input')));
    expect(wrapper.find('input')).toHaveLength(1);
    expect(wrapper.find('button')).toHaveLength(1);
  });
});
