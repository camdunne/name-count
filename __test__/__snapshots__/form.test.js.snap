// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders correctly 1`] = `
<center>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="xmlfile"
    >
      Enter an xml file:  
      <input
        name="xmlfile"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <button
      id="submit"
      type="submit"
    >
      Submit
    </button>
  </form>
</center>
`;
